libraries/core_libs/consensus/src/transaction/transaction_manager.cpp

```diff
@@ -349,4 +354,18 @@ bool TransactionManager::checkBlockTransactions(DagBlock const &blk) {
   return true;
 }
 
+void TransactionManager::blockFinalized(uint64_t block_number) {
+  std::unique_lock transactions_lock(transactions_mutex_);
+  transactions_pool_.blockFinalized(block_number);
+  auto &insufficient_balance_transactions = transactions_pool_.getInsufficientBalanceTransactions();
+  for (auto it = insufficient_balance_transactions.begin(); it != insufficient_balance_transactions.end();) {
+    if (verifyTransaction(it->second.second).first == TransactionStatus::Verified) {
+      it = insufficient_balance_transactions.erase(it);
+      transactions_pool_.insert({it->second.second, TransactionStatus::Verified});
+    } else {
+      it++;
+    }
+  }
```

Comment:
this is ugly, let's rather send `verifyTransaction` as std::function arg and do the whole thing inside the pool, also get rid of the `getInsufficientBalanceTransactions`

Reply:
I  do not think we can actually do this, this way are re-validating 1000 trx for 10 blocks :) I guess we should not do it at all. This is place for attack that we multiply by ourselfs

Reply:
I think that we should. Maybe the number 1000 is too much. But I think this needs to be handled on setting node malicious if they send too many of these transactions in certain time limit. But I would create a separate issue for that as part of general ddos protection.